src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1239,7 +1235,7 @@ async ValueTask<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, b
                         reader.Init(this, behavior, _statements, sendTask);
                         connector.CurrentReader = reader;
                         if (async)
-                            await reader.NextResultAsync(cancellationToken);
+                            await reader.NextResultAsync(CancellationToken.None);
```

Comment:
Is this intentional?

Reply:
Yes - there is a cancel request registered on the cancellation token. So, if we pass the token below, we would never read the PG cancellation error. While the fix is simple, I would like for it to be in a separate pr.

Reply:
OK, no problem!